### DISCUSSION TITLE
Implicit serialization

### DISCUSSION THREAD

**Discussion by sualko** [mikro-orm/mikro-orm#5192] [General]
I have a question about the documentation https://mikro-orm.io/docs/serializing#implicit-serialization

It's not completely clear to me if `it works entirely based on populate hints` is relating only to the initial populate hints of the `find*` method, or if it relates to all `populate` methods. So basically if I populate some collections afterwards, I currently have to call `populated()` on those as well. Is this intended?

```
const author = em.find(Author, ID, {populate: ['books']);

for (const book of books) {
  book.populate(['tags']);

  book.tags.populated(); // if I remove this line, tags is not part of the result of `author.toObject()`.
}
```

**Comment by B4nan** [comment]
I think the problem with your approach is that the root of the serialization context is not aware of your inner populate call - not to mention this is an antipattern as you are introducing N+1 problem this way.

Any reason you don't do this instead?

```ts
const author = await em.find(Author, ..., {populate: ['books.tags']);
```

Or separately at a later stage via `em.populate`, but still on the root entity, which I guess for you is the author:

```ts
const author = await em.find(Author, ..., {populate: ['books']);
await em.populate(author, ['books.tags']);
```

Anyway, in cases like this, it's often better to just use explicit serialization.

**Reply by sualko** [reply]
> this is an antipattern as you are introducing N+1 problem this way

This was only an example. It is clear that there are better options in this case. It was more about the general functionality, but as I understand you, this behavior is intended. Maybe it would be helpful to clarify this in the documentation, because if you know it you can work around it as shown.
